addon/webextension/background/communication.js

```diff
@@ -18,6 +18,7 @@ window.communication = (function () {
     let func = registeredFunctions[req.funcName];
     let result;
     try {
+      req.args.push(sender);
```

Comment:
We'll just have to be sure that we don't have any functions with implicitly optional last arguments.  Which, on consideration... we do!  `sendEvent` works this way.

We could overload `this` with the sender, but that's kind of a lame form of optional argument.  We could be very careful about how sendEvent works and look for any object arguments where we expect strings, though there's an optional options argument at the end of sendEvent that is an object.  We could also change sendEvent in the selector process to always send the expected number of arguments.  I'm not sure what the best approach is.